StoreForecasts/Program.cs

```diff
@@ -31,29 +31,83 @@ class Program {
+      var fcsts = await GetNewForecastRecords();
+
+      var recordsStored = await StoreNewRecords(fcsts);
 
-      // var (nFcstRecordsStored, nFcstValueRecordsStored) = await StoreNewRecords(fcstRecords, fcstValueRecords);
-      // Console.WriteLine($"Wrote {nFcstRecordsStored} forecast records to the database");
-      // Console.WriteLine($"Wrote {nFcstValueRecordsStored} forecast value records to the database");
+      Console.WriteLine($"Wrote {recordsStored} records to the database");
     }
 
-    record ForecastData(IEnumerable<LoadForecast> FcstRecords, IEnumerable<LoadForecastValue> FcstValueRecords) { }
+    static async Task<IEnumerable<LoadForecast>> GetNewForecastRecords() {
+      var opAreas = await GetOpAreas();
+      var opAreaIDsByName = opAreas.ToDictionary(keySelector: oa => oa.Name, elementSelector: oa => oa.Id);
+      var newApiFcstRecords = await GetApiForecasts(opAreas);
+      return from newRecord in newApiFcstRecords select ApiRecordToDbRecords(newRecord, opAreaIDsByName);
+    }
+
+    static async Task<List<ForecastJson>> GetApiForecasts(IEnumerable<OpArea> opAreas) {
+      var maxDate = DateOnly.FromDateTime(DateTime.Today);
+
+      var allForecasts = new List<ForecastJson>();
+      foreach (var opArea in opAreas) { 
+        var latestDate = await LoadLatestDateFromDatabase(opArea.Id);
+        var minDate = latestDate?.AddDays(1) ?? StartDate;
+
+        var query = new Dictionary<string, string> {  
+          ["operatingArea"] = opArea.Name,
+          ["startDate"] = minDate.ToShortDateString(),
+          ["endDate"] = maxDate.ToShortDateString(),
+          ["idf"] = "Nom Plan",
+        };
+        var uri = QueryHelpers.AddQueryString($"https://{MCastDomain}/api/v1/daily/forecasted-load", query);
+        var response = await Client.GetAsync(uri);
+        Console.WriteLine(response.ToString());
+        response.EnsureSuccessStatusCode();
+        var json = await response.Content.ReadAsStringAsync();
+        var forecasts = JsonConvert.DeserializeObject<List<ForecastJson>>(json)!;
+        allForecasts.AddRange(forecasts);
+      }
+
+      return allForecasts;
+    }
 
-    static async Task<ForecastData> GetNewForecastRecords() {
-      throw new NotImplementedException();
+    static LoadForecast ApiRecordToDbRecords(ForecastJson json, IDictionary<string, short> opAreaIDsByName) {
+      var fcstRecord = 
+        new LoadForecast { 
+          Date = json.ForecastStartDate.ToDateTime(TimeOnly.MinValue),
+          OpArea = opAreaIDsByName[json.OperatingArea],
+        };
+
+      var valueRecords = 
+        ( 
+          from value in json.LoadForecast 
+          select new LoadForecastValue {
+            Horizon = (byte)value.DaysOut,
+            Value = value.Forecast,
+            ForecastNavigation = fcstRecord,
+          }
+        ).ToList();
+
+      fcstRecord.LoadForecastValues = valueRecords;
+
+      return fcstRecord;
```

Comment:
Kinda cool that I can have LoadForecast in the DB with an IDENTITY column, and then define this FK, and then in code I'm not messing with ID values at all.  I just set this `ForecastNavigation` field in the `LoadForecastValue` records, and then assign the `LoadForecastValues` field of the `LoadForecast` record, and then when I insert the fcst record, it also inserts all the values and links them by ID.